go/vt/tabletserver/tabletconn/tablet_conn.go

```diff
@@ -138,19 +121,54 @@ type TabletConn interface {
 		algorithm querypb.SplitQueryRequest_Algorithm) (queries []querytypes.QuerySplit, err error)
 
 	// StreamHealth starts a streaming RPC for VTTablet health status updates.
```

Comment:
This needs a much more detailed description of the `sendReply` function.

Missing bits are:
- if the callback returns a non-nil error, the error will be forwarded to the caller and the stream will be closed
- the only exception is io.EOF: in that case the stream will be closed but the caller will see a nil error